### PR TITLE
Carousel buttons

### PR DESCRIPTION
**Pull request by scottMounce**
@Stuzerain Good evening! Check out how my buttons are working! I have them setup so if you click next or back it goes through a loop (back to start if you're at the last index, and to the end if you are at the first index and clicking back). Also I have everything filtering by 4! 

https://recordit.co/mc0WUSbN8v

### CODE REVIEW

**Comment by Stuzerain** [comment]
@scottMounce Looking good! Definitely coming together. My notes would be that there are a few places you could probably consolidate your code a little bit moving forward, but these would be things to 'stretch' toward if you have time. 

For example, your onNextPageButtonClick and onBackPageButtonClick could probably be the same function! I pared my navigation buttons into one function that does either thing depending on a 'direction' argument it gets passed. That way, You can keep that function in one place and you can minimize the number of props you're passing down for functionality.

Also, this is a stylistic choice that I like, but I tend to try to keep my rendering/returning as concise as I can. In the PageCarouselButtons.jsx component, your return looks like this: 

  return (    
    <div>
      <button onClick={(e) => {
        onBackPageButtonClick(e, arrOfFours[index -1], index -1);
      }}>Back</button>
      {arrOfFours.map((pageButtons) => {          
        ++buttonNum;
        return <CarouselButton
        onPageNumberButtonClick={onPageNumberButtonClick}
        arrOfFours = {arrOfFours}
        pageButtons={pageButtons}
        buttonNum={buttonNum}
        key={buttonNum}
        />
      })}
      <button onClick={(e) => {
        onNextPageButtonClick(e, arrOfFours[index + 1], index + 1);
      }}>
        Next
        </button>
    </div>
  )

Your 'arrOfFours.map()' in there can make it a little hard to follow at a glance. What I like to do is move something like that into a variable outside the return, like this:

 const numberedButtons = {arrOfFours.map((pageButtons) => {          
        ++buttonNum;
        return <CarouselButton
        onPageNumberButtonClick={onPageNumberButtonClick}
        arrOfFours = {arrOfFours}
        pageButtons={pageButtons}
        buttonNum={buttonNum}
        key={buttonNum}
        />
      })}

  return (    
    <div>
      <button onClick={(e) => {
        onBackPageButtonClick(e, arrOfFours[index -1], index -1);
      }}>Back</button>
     {numberedButtons}
      <button onClick={(e) => {
        onNextPageButtonClick(e, arrOfFours[index + 1], index + 1);
      }}>
        Next
        </button>
    </div>
  )

Totally optional, but just a thought! Keep up the good work, and merge away!

**Comment by scottMounce** [comment]
@Stuzerain Thanks! I appreciate your feedback, I think if I have extra time I can for sure refactor some of the code to have less code reuse!
